packages/programs/src/DefaultProgramSystem.sol

```diff
@@ -22,13 +22,23 @@ contract DefaultProgramSystem is System {
     return createAccessGroup(owner);
   }
 
-  function setAccessGroup(EntityId caller, EntityId target, uint256 groupId) external {
+  function setAccessGroup(EntityId caller, EntityId target, uint256 groupId) public {
     caller.validateCaller();
-    uint256 currentGroupId = EntityAccessGroup.get(target);
+    (uint256 currentGroupId,) = getGroupId(target);
     _requireOwner(currentGroupId, caller);
     EntityAccessGroup.set(target, groupId);
   }
 
+  /// @dev This function can be called by other entities to get a new access group assigned to themselves
+  function setAccessGroup(EntityId target, address groupOwner) external {
```

Comment:
```suggestion
  function setAccessGroup(EntityId caller, address groupOwner) external {
```
would this be more accurate?

Reply:
I was just trying to make it a bit clearer that it is setting the group for itself